src/Hook/ProcessFormDataHook.php

```diff
@@ -103,13 +104,18 @@ public function __invoke(array $postData, array $formConfig, array|null $arrFile
      */
     private function generateOptInUrl(string $token, array $formConfig): string
     {
-        $page = $GLOBALS['objPage'];
+        $url = Environment::get('uri');
 
         if ($formConfig['leadOptInTarget']) {
             $page = PageModel::findWithDetails($formConfig['leadOptInTarget']);
+
+            try {
+                $url = $page->getAbsoluteUrl();
+            } catch (\Exception $e) {
+                // Do nothing
+            }
         }
 
-        $url = $page->getAbsoluteUrl();
         $parameter = '?token='.$token;
```

Comment:
As we now use the uri from the environment (which I think is not a bad idea) don't we have to check if there are already url parameters in the url and depending on that append the token with '?' or '&'?

Reply:
Good point, fixed the issues and also added a uri parameter check.